enterprise_learner_portal/api/v1/serializers.py

```diff
@@ -81,3 +84,17 @@ def _get_course_overview(self, course_run_id):
                 return overview
 
         return None
+
+    def _get_course_run_url(self, request, course_run_id):
+        """
+        Get the appropriate course url while incorporating Executive Education content.
+        """
+        course_key = CourseKey.from_string(course_run_id)
+        course_key_str = '{}+{}'.format(course_key.org, course_key.course)
+        course_data = get_course_catalog_api_service_client().get_course_details(course_key_str)
```

Comment:
B2B and B2C use different approaches for building their urls ... B2B uses course_run_url and B2C utilizes the learner's progress. Hence, we would need separate updates to the codebase.

Reply:
@irfanuddinahmad Can you please clarify with specific links? What do you mean by "B2C utilizes the learner's progress"? 

Looking at the code, both the B2B `enterprise-course-enrollments` endpoint and the B2C http://localhost:18000/api/learner_home/init endpoint serving home.edx.org largely utilize the same helper function `course_home_url` in edx-platform ([source](https://github.com/openedx/edx-platform/blob/ba8a3c130db5936bf0851b8db261b6c7499968e0/openedx/features/course_experience/__init__.py#L100-L108)).

The only difference is that the B2B `enterprise-course-enrollments` API gets the `course_home_url` via another B2C helper function `get_course_run_url` in edx-platform ([source](https://github.com/openedx/edx-platform/blob/ba8a3c130db5936bf0851b8db261b6c7499968e0/lms/djangoapps/course_api/api.py#L281-L293)) whereas the B2C learner home `init` API calls `course_home_url` directly ([source](https://github.com/openedx/edx-platform/blob/ba8a3c130db5936bf0851b8db261b6c7499968e0/lms/djangoapps/learner_home/serializers.py#L114-L115)).

Given these 2 endpoints are already pulling from the same `course_home_url`, it would seem reasonable to implement a solution in edx-platform to conditionally render the GetSmarter dashboard url there (in an open-source friendly way) such that no changes need to be made to the `enterprise-course-enrollments` API regarding the course run URL and the solution would be applicable to the B2C dashboard already.

Reply:
Reverting this PR for implementing the changes in [course_home_url](https://github.com/openedx/edx-platform/blob/ba8a3c130db5936bf0851b8db261b6c7499968e0/openedx/features/course_experience/__init__.py#L100) in edx-platform
The B2C dashboard uses the course_home_url from [here](https://github.com/openedx/edx-platform/blob/ba8a3c130db5936bf0851b8db261b6c7499968e0/lms/djangoapps/learner_home/serializers.py#L114) and the B2B dashboard uses it from [here](https://github.com/openedx/edx-platform/blob/ba8a3c130db5936bf0851b8db261b6c7499968e0/lms/djangoapps/course_api/api.py#L293)